src/ray/util/memory.cc

```diff
@@ -27,7 +29,8 @@ uint8_t *pointer_logical_and(const uint8_t *address, uintptr_t bits) {
 
 void parallel_memcopy(uint8_t *dst, const uint8_t *src, int64_t nbytes,
                       uintptr_t block_size, int num_threads) {
-  std::vector<std::thread> threadpool(num_threads);
+  static auto &threadpool =
```

Comment:
why do we need & here?

Reply:
I don't think we need to.

Reply:
@mwtian

Reply:
Using a reference matches what is recommended in google style guide: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables. Otherwise it needs to be a const pointer which is more clunky. Storing by value is not recommended because its destructor may run during shutdown, concurrently with other logic.

Reply:
It is based on `If all else fails, you can create an object dynamically and never delete it by using a function-local static pointer or reference (e.g., static const auto& impl = *new T(args...);).` right?

Reply:
@rkooo567 Yes.